ArduPlane/GCS_Mavlink.cpp

```diff
@@ -973,22 +976,28 @@ MAV_RESULT GCS_MAVLINK_Plane::handle_command_int_guided_slew_commands(const mavl
 
         float new_target_heading = radians(wrap_180(packet.param2));
 
-        // course over ground
-        if ( int(packet.param1) == HEADING_TYPE_COURSE_OVER_GROUND) { // compare as nearest int
+        switch(HEADING_TYPE(packet.param1)) {
+        case HEADING_TYPE_COURSE_OVER_GROUND:
+            // course over ground
             plane.guided_state.target_heading_type = GUIDED_HEADING_COG;
             plane.prev_WP_loc = plane.current_loc;
-        // normal vehicle heading
-        } else if (int(packet.param1) == HEADING_TYPE_HEADING) { // compare as nearest int
+            break;
+        case HEADING_TYPE_HEADING:
+            // normal vehicle heading
             plane.guided_state.target_heading_type = GUIDED_HEADING_HEADING;
-        } else {
+            break;
+        case HEADING_TYPE_DEFAULT:
+            plane.guided_state.target_heading_type = GUIDED_HEADING_NONE;
+            return MAV_RESULT_ACCEPTED;
+        default:
             //  MAV_RESULT_DENIED  means Command is invalid (is supported but has invalid parameters).
             return MAV_RESULT_DENIED;
         }
 
         plane.g2.guidedHeading.reset_I();
 
         plane.guided_state.target_heading = new_target_heading;
-        plane.guided_state.target_heading_accel_limit = MAX(packet.param3, 0.05f);
+        plane.guided_state.target_heading_accel_limit = MAX(is_zero(packet.param3)? 10.0f : packet.param3 , 10.0f); // the, previous limit of 0.05 was 0.29 degrees, not very useful
```

Comment:
Why this change?

Reply:
As per the comment @iampete // the, previous limit of 0.05 was 0.29 degrees, not very useful
The value is in radians so It was throttling the rate at which heading would change to very, very tiny increments that prevented the follow from  being able to track the target plane.

Reply:
It's a max not a min, it's saying the lowest it can be is  0.05 but it can be larger.

Reply:
> It's a max not a min, it's saying the lowest it can be is 0.05 but it can be larger.

Read that again - does it make sense to you?

Reply:
> Read that again - does it make sense to you?

Yes, but then I'm dyslexic. Let me try again. 

Its a `MAX` operation. The output can be anything larger than 0.05.

Your complaint seems to be that 0.05 is too small? The `MAX` does not prevent you using a larger value.